### PR TITLE
fix: do not append duplicates to user agent string

### LINKED ISSUE #565: Use of Batch adds duplicate, redundant gcloud-python/x.xx.x to User-Agent header in an unbounded fashion
### Overview
We first became aware of this issue when we began seeing "Error 413 (Request Entity Too Large)" responses to a variety of GCS requests after we implemented blob.reload() calls in a Batch context.   We discovered after capturing the HTTP requests that the User-Agent headers for both the "outer" batch request and the contained requests were all filled with hundreds of 'gcloud-python/1.40.0' specifications.  For example, we've seen a single batch request containing 51 reload requests with each   User-Agent specification containing over 100 repetitions of 'gcloud-python/1.40.0'.  Overall, this request had 5,300 occurrences of the same 'gcloud-python/1.40.0' 

#### Analysis
When batch calls are made, a ' gcloud-python/x.xx.x ' is blindly appended to the underlying connection's _client_info.user_agent string:

https://github.com/googleapis/python-storage/blob/master/google/cloud/storage/_http.py#L59

As a result, successive batch calls result in the connection._client_info.user-agent string accumulating more and more, duplicate ' gcloud-python/x.xx.x ' specifications.  If a Client object continues making batch requests, the user-agent property continues to grow without limit.

#### Environment details
  - OS type and version: Ubuntu 20.04
  - Python version: 3.7.11
  - pip version: 21.2.4
  - `google-cloud-storage` version: 1.42.0

#### Repro Script
```
from google.api_core import exceptions as g_exc
from google.cloud.storage.blob import Blob
from google.cloud.storage.client import Client

project = '<GCP PROJCT NAME>'
bucket_name = '<GCS BUCKET NAME>'

# Client.__init__ initializes self._connection which initiallizes a Connection object as defined in
# google.cloud.storage._http.py which automatically appends a 'gcloud-python/x.xx.x' spec to the
# the user_agent property
c = Client(project=project)
print(c._connection._client_info.user_agent)

bucket = c.get_bucket(bucket_name)
blobs = list(c.list_blobs(bucket))[:1]

# ----- First batch call adds duplicate user_agent spec
print("Before batch: "  + c._connection._client_info.user_agent)

try:
    # Creation of a Batch object results in another call which again appends another (duplicate) 
    # 'gcloud-python/x.xx.x' to the user_agent property.
    with c.batch()as batch:
        print("Batch context: " + batch._client_info.user_agent)
        for blob in blobs:
            blob.reload()
except g_exc.NotFound:
    pass
print("After batch: "  + c._connection._client_info.user_agent)

# ----- Subsequent batch calls continue to add more duplicates ad nauseam
for i in range(5):
    try:
        with c.batch()as batch:
            print(f"Iteration {i} - Batch context: " + batch._client_info.user_agent)
            for blob in blobs:
                blob.reload()
    except g_exc.NotFound:
        pass

```

#### Repro instructions
  1. Modify repro script to specify valid 'project' and 'bucket_name' values
  2. Execute script

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #565 🦕

### CODE REVIEW

File: google/cloud/storage/_http.py [hunk 1]
```diff
@@ -56,7 +56,8 @@ def __init__(self, client, client_info=None, api_endpoint=None):
         # TODO: When metrics all use gccl, this should be removed #9552
         if self._client_info.user_agent is None:  # pragma: no branch
             self._client_info.user_agent = ""
-        self._client_info.user_agent += " gcloud-python/{} ".format(__version__)
+        if "gcloud-python/{}".format(__version__) not in self._client_info.user_agent:
+            self._client_info.user_agent += " gcloud-python/{} ".format(__version__)
```
Comment (tseaver) [inline google/cloud/storage/_http.py:60]:
```suggestion
        agent_version = "gcloud-python/{}".format(__version__)
        if  agent_version not in self._client_info.user_agent:
            self._client_info.user_agent += " {} ".format(agent_version)
```

**Review by tseaver — CHANGES_REQUESTED** [review]
@ddelgrosso1 Can you please add a unit test which reproduces the issue (i.e., failing with this change not applied, but passing with it applied)?

**Comment by tseaver** [comment]
@gooblebot I consent.

**Comment by tseaver** [comment]
@ddelgrosso1 The `lint` session is failing, likely to do with something in the suggestion you merged.  Can you pull from here to your local branch and run `nox -s blacken` and commit / push the resulting change?

**Comment by andrewsg** [comment]
Denis, please ping me if you would like help running the auto-formatter/linter and writing the regression test.

**Review by tseaver — APPROVED** [review]
@ddelgrosso1 Thanks for the test!  I verified that it fails against `main` without the patch.

**Pull request merged**